### PR TITLE
Add ETH-based Swap and Cross-Chain Functionality

### PR DESCRIPTION
**Pull request by SibghatUllah1997**
**Description of Changes**
This pull request introduces the `swapAndCrossOneInchETH` function to our smart contract suite. This addition enables users to perform ETH-based swaps and initiate cross-chain transactions seamlessly. The function is designed to accept Ethereum as input and interact with the 1inch protocol for efficient swapping and cross-chain operations.

**Key Features**
- Accepts Ethereum (ETH) as the input token and handles conversion to WETH.
- Facilitates local ETH swaps and prepares for cross-chain transactions.
- Implements comprehensive input validation checks to ensure transaction integrity and security.
- Emits detailed events for better tracking and transparency of operations.

**Technical Details**
- The function is marked `external` and `payable` to handle incoming ETH transactions.
- Utilizes `IWETH` interface for WETH interactions.
- Detailed NatSpec comments added for better understanding and documentation purposes.
- Includes prefixed `require` statements for clear and informative error handling.

This function enriches our contract's capabilities, making it more versatile in handling various types of token swaps and cross-chain interactions. The addition follows our commitment to enhancing user experience and expanding the functionality of our DeFi solutions.

### CODE REVIEW

File: contracts/upgradeable-Bridge/FiberRouter.sol [hunk 1]
```diff
@@ -16,6 +17,7 @@ contract FiberRouter is Ownable, TokenReceivable {
     using SafeERC20 for IERC20;
     address public pool;
     address public oneInchAggregatorRouter;
+    address public constant WETH = 0xc778417E063141139Fce010982780140Aa0cD5Ab; //need to change for every blockchain
```
Comment (naiemk) [inline contracts/upgradeable-Bridge/FiberRouter.sol:20]: Don't use this. See comments

**Review by naiemk — COMMENTED** [review]
You need to pass the WETH address as constructor parameters. Better option will be to use the `Deployer` pattern where you deploy the contract using `FerrumDeployer` and pass the parameters through the `initData` (`IFerrumDeployer(msg.sender).initData()`).

To see how to use the Ferrum Deployer follow this pattern:

Check this contract [for example](https://github.com/ferrumnet/quantum-portal-smart-contracts/blob/main/contracts/quantumPortal/poc/QuantumPortalGateway.sol)

Line 10, import the deployer
Line 27, get the init params
Line 28, decode the init params. For example, you can pass `pool`, `oneInchRouter` and `WETH` as init params. Obviously these params must be abi encoded and passed to the deployer.

2. Create a dummy file [this](https://github.com/ferrumnet/quantum-portal-smart-contracts/blob/main/contracts/quantumPortal/ReferencesForTest.sol) and re-instatiate the FerrumDeployer. This is so that hardhat pulls the code for the contract.

3. Update the deploy script:
Follow [this](https://github.com/ferrumnet/quantum-portal-smart-contracts/blob/main/scripts/quantumPortal/poc/examples/deployMultiChainToken.ts#L11) example in the deploy script. 
Line 11 encodes the init parameters and line 12 deploys the contract.

To import the `deployUsingDeployer` function you can reference the `foundry-contracts` package:

```
import { deployUsingDeployer, Salt, ZeroAddress } from "foundry-contracts/dist/test/common/Utils";
```

File: contracts/upgradeable-Bridge/FiberRouter.sol (hunk 1 above)
Comment (taha-abbasi) [reply contracts/upgradeable-Bridge/FiberRouter.sol:20]: Thanks @naiemk 

@SibghatUllah1997 please see the approach above. Share if you have any questions.

**Review by taha-abbasi — APPROVED** [review]
https://ferrumnetwork.slack.com/archives/C01UMJSBTMH/p1706000088235379?thread_ts=1702318662.886489&cid=C01UMJSBTMH

**Pull request merged**